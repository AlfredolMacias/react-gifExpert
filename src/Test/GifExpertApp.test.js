import React from 'react';
const { shallow } = require("enzyme");
const { GifExpertpp } = require("../GifExpertpp");


describe('Pruebas con el componente <GifExpertApp />', () => {
    
    
    test('Debe coincidir el snapshot', () => {
        const wrapper = shallow( <GifExpertpp />)
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe mostrar una lista de categorias', () => {
        
        const categories = ['The Office', 'Brooklyn 99'];
        const wrapper = shallow( <GifExpertpp defaultCategories= { categories } />);
        
        expect ( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
        
    })
    


    
})
