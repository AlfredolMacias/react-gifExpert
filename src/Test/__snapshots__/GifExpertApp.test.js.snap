// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifExpertApp /> Debe coincidir el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertpp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con el componente <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertpp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Office"
      key="The Office"
    />
    <GifGrid
      category="Brooklyn 99"
      key="Brooklyn 99"
    />
  </ol>
</Fragment>
`;
